src/utilities/io.jl

```diff
@@ -175,81 +137,35 @@ function flatten(names, value :: Array{Float64}, k :: String, v)
     return
 end
 
-function Base.getindex(c::Chain, v::Symbol)
-    # This strange implementation is mostly to keep backward compatability.
-    #  Needs some refactoring a better format for storing results is available.
-    if v == :logevidence
-        return log(c.weight)
-    elseif v==:logweights
-        return c[:lp]
-    else
-        idx = names2inds(c, string(v))
-        if any(idx .== nothing)
-            syms = collect(Iterators.filter(k -> occursin(string(v)*"[", string(k)), c.names))
-            sort!(syms)
-            idx = names2inds(c, syms)
-            @assert all(idx .!= nothing)
-            return c.value[:, idx, :]
-        else
-            return getindex(c, string(v))
-        end
-    end
-end
-
-function Base.getindex(c::Chain, v::String)
-    return c.value[:, names2inds(c, v), :]
-end
-
-
-function Base.getindex(c::Chain, expr::Expr)
-    str = replace(string(expr), r"\(|\)" => "")
-    @assert match(r"^\w+(\[(\d\,?)*\])+$", str) != nothing "[Turing.jl] $expr invalid for getindex(::Chain, ::Expr)"
-    return c[Symbol(str)]
-end
-
-function Base.vcat(chains::Chain...)
-
-    c1 = chains[1]
-
-    weight = c1.weight
-    names = c1.names
-    range = c1.range
-    chains = c1.chains
-
-    value = c1.value
-    info = c1.info
-
-    for c in chains[2:end]
-        all(c -> c.names == names, args) ||
-            throw(ArgumentError("chain names differ"))
-
-        all(c -> c.chains == chains, args) ||
-            throw(ArgumentError("sets of chains differ"))
-
-        @assert c1.weight == c.weight
-        @assert c1.range == c.range
-
-        value = cat(value, c.value, dims=1)
-        info = merge(info, c.info)
-    end
-
-    chn = Chain(weight,
-                value,
-                range,
-                names,
-                chains,
-                info)
+function Base.vcat(c1::Chains, args::Chains...)
```

Comment:
This should have not been changed back to the old version which is brocken. Seem like you are accidentally overwriting master with an old version here..

Reply:
Good catch, thank you! Updated.